core/src/components/router/utils/parser.ts

```diff
@@ -17,46 +27,33 @@ export const readRoutes = (root: Element): RouteChain[] => {
   return flattenRouterTree(readRouteNodes(root));
 };
 
-export const readRouteNodes = (root: Element, node = root): RouteTree => {
+export const readRouteNodes = (node: Element): RouteTree => {
   return (Array.from(node.children) as HTMLIonRouteElement[])
     .filter(el => el.tagName === 'ION-ROUTE' && el.component)
     .map(el => {
-      const component = readProp(el, 'component');
-      if (component == null) {
-        throw new Error('component missing in ion-route');
-      }
```

Comment:
Do we still need this error?

Reply:
Look at the `filter()` on l 22 above, `el.component` is guaranteed not to be null then `readProp(el, 'component')` could not be null